### PR TITLE
Ship cells and state

### PR DESCRIPTION
**Pull request by maxnation**
shipCells and ship showState()

### CODE REVIEW

File: SeaFightLogic/Ships/Ship.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        public int ActionDistance { get; protected set; }
+        public int Size { get; protected set; }
+        public bool[] ShipCells { get; set; }
+
+        public Ship(int size, int speed)
+        {
+            int maxSize = Convert.ToInt32(ConfigurationManager.AppSettings["maxShipSize"]);
+            if (size < 1 || size > maxSize)
+            {
+                throw new ArgumentOutOfRangeException("Size of ship must be from 1 to 5");
```
Comment (evavsun) [inline SeaFightLogic/Ships/Ship.cs]: maxSize instead of hardcoded "5"

File: SeaFightLogic/Ships/Ship.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            {
+                ShipCells[i] = true;
+            }
+
+            Size = size;
+        }
+
+        public void ShowState()
+        {
+            Console.WriteLine("Ship type: {0}", this.GetType().Name);
```
Comment (evavsun) [inline SeaFightLogic/Ships/Ship.cs]: It doesn't matter where current service can show the state

File: SeaFightLogic/Ships/Warship.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        public void Shoot(int distance)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Warship(int size, int speed, int actionDistance) : base(size, speed)
+        {
+            int maxDistance = Convert.ToInt32(ConfigurationManager.AppSettings["warShipDistance"]);
+
+            if (actionDistance < 1 || actionDistance > 5)
```
Comment (evavsun) [inline SeaFightLogic/Ships/Warship.cs]: maxDistance  instead of "5"

File: SeaFightLogic/Ships/Ship.cs (hunk 1 above)
Comment (maxnation) [reply SeaFightLogic/Ships/Ship.cs]: done

File: SeaFightLogic/Ships/Ship.cs (hunk 2 above)
Comment (maxnation) [reply SeaFightLogic/Ships/Ship.cs]: Now return Dictionary<string,string>. But must replace with json object later

File: SeaFightLogic/Ships/Warship.cs (hunk 3 above)
Comment (maxnation) [reply SeaFightLogic/Ships/Warship.cs]: done

**Pull request merged**